app/controllers/sponsors_controller.rb

```diff
@@ -0,0 +1,55 @@
+class SponsorsController < ApplicationController
+  before_action :set_sponsor, only: [:show, :edit, :update, :destroy]
+  before_action :get_event
```

Comment:
use 1 convention for these callbacks like we are using `set` for sponsor but `get` for event.